unbound/Dockerfile-unbound

```diff
@@ -1,47 +1,76 @@
-FROM ubuntu:18.04
```

Comment:
Overall, the Dockerfile and the shell script are well written and follow best practices. However, there are a few points that could be improved:

1. **Use of latest tag**: The code uses the `latest` tag for the Unbound download URL. It is generally recommended to avoid using the `latest` tag and instead pin versions in dockerfiles to make builds reproducible.

```diff
- ENV UNBOUND_DOWNLOAD_URL https://www.unbound.net/downloads/unbound-latest.tar.gz
+ ENV UNBOUND_DOWNLOAD_URL https://www.unbound.net/downloads/unbound-1.13.2.tar.gz
```

2. **Use of `apk upgrade`**: The `apk upgrade` command is used which is generally not recommended in Dockerfiles as it can lead to unnecessary size increase and potential instability.

```diff
- RUN apk update && apk upgrade
+ RUN apk update
```

3. **Grouping of `RUN` instructions**: To reduce the number of layers in the image, it is recommended to group `RUN` instructions together.

```diff
- RUN apk update && apk add build-base linux-headers \
-            libtool automake autoconf cmake ninja \
-            make \
-            openssl openssl-dev \
-            git wget
- RUN git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs && \
-    git clone --depth 1 --branch master https://github.com/openssl/openssl.git && \
-    git clone --depth 1 --branch main https://github.com/open-quantum-safe/oqs-provider.git
+ RUN apk update && \
+     apk add build-base linux-headers libtool automake autoconf cmake ninja make openssl openssl-dev git wget && \
+     git clone --depth 1 --branch main https://github.com/open-quantum-safe/liboqs && \
+     git clone --depth 1 --branch master https://github.com/openssl/openssl.git && \
+     git clone --depth 1 --branch main https://github.com/open-quantum-safe/oqs-provider.git
```

4. **Use of `chmod 777`**: Using `chmod 777` is generally not recommended as it gives all permissions to all users. Consider using more restrictive permissions.

```diff
- RUN chmod 777 /opt/unbound.sh
+ RUN chmod 755 /opt/unbound.sh
```

5. **Cleaning up**: After installing packages, it's a good practice to clean up the cache to reduce the size of the image.

```diff
+ RUN rm -rf /var/cache/apk/*
```

6. **Use of `set -x`**: The `set -x` command is used which prints all commands before executing them. This could potentially expose sensitive information in logs. Consider removing it if not necessary.

```diff
- RUN set -x && \
+ RUN \
```

7. **Use of `wget --no-check-certificate`**: It's generally not a good practice to disable certificate check. Consider removing `--no-check-certificate` if the server's SSL certificate is valid.

```diff
- wget --no-check-certificate -O unbound.tar.gz $UNBOUND_DOWNLOAD_URL && \
+ wget -O unbound.tar.gz $UNBOUND_DOWNLOAD_URL && \
``